coreneuron/nrniv/cn_parameters.cpp

```diff
@@ -0,0 +1,171 @@
+AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+THE POSSIBILITY OF SUCH DAMAGE.
+*/
+
+#include "coreneuron/nrniv/cn_parameters.h"
+
+cn_parameters::cn_parameters(){
+
+    app.get_formatter()->column_width(50);
+    app.set_help_all_flag("-H, --help-all", "Print this help including subcommands and exit.");
+
+    app.set_config("--config", "config.ini", "Read parameters from ini file", false)
+        ->check(CLI::ExistingFile);
+    app.add_flag("--mpi", this->mpi_en, "Enable MPI. In order to initialize MPI environment this argument must be specified." );
+    app.add_flag("--gpu", this->gpu, "Activate GPU computation.");
+    app.add_option("--dt", this->dt, "Fixed time step. The default value is set by defaults.dat or is 0.025.", true)
+        ->check(CLI::Range(-1000.,1e9));
+    app.add_option("-e, --tstop", this->tstop, "Stop Time in ms.")
+        ->check(CLI::Range(0., 1e9));
+    app.add_flag("--show", this->print_arg, "Print arguments.");
+
+    auto sub_gpu = app.add_subcommand("gpu", "Commands relative to GPU.");
+    sub_gpu -> add_option("-W, --nwarp", this->nwarp, "Number of warps to balance.", true)
+        ->check(CLI::Range(0, 1000000));
+    sub_gpu -> add_option("-R, --cell-permute", this->cell_interleave_permute, "Cell permutation: 0 No permutation; 1 optimise node adjacency; 2 optimize parent adjacency.", true)
+        ->check(CLI::Range(0, 3));
+
+    auto sub_input = app.add_subcommand("input", "Input dataset options.");
+    sub_input -> add_option("-d, --datpath", this->datpath, "Path containing CoreNeuron data files.")
+        ->check(CLI::ExistingPath);
+    sub_input -> add_option("-f, --filesdat", this->filesdat, "Name for the distribution file.", true)
+        ->check(CLI::ExistingFile);
+    sub_input -> add_option("-p, --pattern", this->patternstim, "Apply patternstim using the specified spike file.")
+        ->check(CLI::ExistingFile);
+    sub_input -> add_option("-s, --seed", this->seed, "Initialization seed for random number generator.")
+        ->check(CLI::Range(0, 100000000));
+    sub_input -> add_option("-v, --voltage", this->voltage, "Initial voltage used for nrn_finitialize(1, v_init). If 1000, then nrn_finitialize(0,...).")
+        ->check(CLI::Range(-1e9, 1e9));
+    sub_input -> add_option("--read-config", this->rconfigpath, "Read configuration file filename.")
```

Comment:
A small naming detail suggestion again. If `rconfigpath` is a file maybe it should be named as `rconfigfilepath` to avoid any confusion between the path where the file is and the file. Same with `restorepath`

Reply:
Fair enough!